### PR TITLE
Update MetaSearch with new settings API

### LINKED ISSUE #54144: add wms data via metasearch causes 'protocol undefined' error
### What is the bug or the crash?

this happens on QGIS 3.30 but used to work on previous versions

image

when I try to add wms layer from a metadata search result in metasearch, the warning pops up 'connection metasearch already exists, overwrite?', when answering yes, this error pops up (from the wms add layer panel) 'protocol undefined'. Seems Metasearch is not properly setting the protocol (and url) of the add wms panel.

image

### Steps to reproduce the issue

- from metasearch run a query against https://www.nationaalgeoregister.nl/geonetwork/srv/eng/csw (use keyword wms)
- browse results until `add data` button is activated
- select a WMS layer to add

- notice the warning: metasearch already exists, overwrite? -> choose yes (or try again)
- notice the error: 'capabilities failed, protocol not defined'

### Versions

QGIS-versie
3.30.2-'s-Hertogenbosch
QGIS code revisie
0992b533
Qt versie
5.15.3
Python versie
3.9.5
GDAL/OGR versie
3.6.4
PROJ versie
9.2.0
EPSG-register databaseversie
v10.082 (2023-02-06)
GEOS versie
3.11.2-CAPI-1.17.2
SQLite versie
3.41.1
PDAL versie
2.5.2
PostgreSQL client versie
unknown
SpatiaLite versie
5.0.1
QWT versie
6.1.6
QScintilla2 versie
2.13.1
OS versie
Windows 10 Version 2009

Actieve Python plug-ins
qgisnetworklogger
0.2.0
quick_map_services
0.19.33
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 3nids**
fixes #54144

### CODE REVIEW

**Comment by 3nids** [comment]
The services seem now to be well configured but I cannot connect to the URL given in the issue. Could someone please test this? @agiudiceandrea ?

File: python/plugins/MetaSearch/dialogs/maindialog.py [hunk 1]
```diff
@@ -718,35 +718,45 @@ def add_to_ows(self):
...
-            stype = ['OGC:WCS', 'wcs', 'wcs']
+            service_type = 'OGC:WCS'
+            sname = 'WCS'
+            dyn_param = ['wcs']
+            provider_name = 'wcs'
+            setting_node = QgsSettingsTree.node('connections').childNode('ows').childNode('connections')
             data_url = item_data['wcs']
         elif caller == 'mActionAddAms':
-            stype = ['ESRI:ArcGIS:MapServer', 'ams', 'arcgismapserver']
+            # TODO: this doesn't exist anymore?
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/dialogs/maindialog.py:743]: Nope, they've all been rolled into the one "arcgis rest services" group

Comment (3nids) [reply python/plugins/MetaSearch/dialogs/maindialog.py]: thanks

**Comment by agiudiceandrea** [comment]
> The services seem now to be well configured but I cannot connect to the URL given in the issue.

@3nids, I can connect to the service https://www.nationaalgeoregister.nl/geonetwork/srv/eng/csw.

Apart from an oversight in the following line:
`            elif service_type = 'ESRI:ArcGIS:FeatureServer':`
which should be changed to:
`            elif service_type == 'ESRI:ArcGIS:FeatureServer':`
it seems to me the patch allows now to correctly add WMS connections and layers via the MetaSearch plugin. I've not tested other type of connections.

**Review by nirvn — APPROVED** [review]

**Comment by pvgenuchten** [comment]
Thanx @3nids, I will test asap

**Pull request merged**